lib/internal/modules/esm/loader.js

```diff
@@ -56,16 +58,32 @@ class Loader {
       await this._resolve(specifier, parentURL, defaultResolve);
 
     if (typeof url !== 'string')
-      throw new ERR_INVALID_ARG_TYPE('url', 'string', url);
+      throw new ERR_INVALID_RETURN_PROPERTY(
+        'string', 'loader resolve', 'url', url
+      );
 
     if (typeof format !== 'string')
-      throw new ERR_INVALID_ARG_TYPE('format', 'string', format);
+      throw new ERR_INVALID_RETURN_PROPERTY_STRING(
+        'string', 'loader resolve', 'format', format
+      );
```

Comment:
This error should likely be `ERR_INVALID_RETURN_PROPERTY` just as the error above. Or I fail to see why this is handled differently than the `url` error case.

Reply:
The first error checks that it is a string, the second error checks that it is the right type of string.

The messages are:

1. Expected string to be returned for the url from the "loader resolve" function but got ${type}.
2. Expected a valid url to be returned for the url from the "loader resolve" function but got ${value}.

Reply:
Ahh, I missed this was specifically the format case, but the same argument applied - `format` must be one of a specific set of string values, so the error message will say:

`Expected a valid format to be returned for the url from the "loader resolve" function but got ${value}`

instead of

`Expected string to be returned for the format from the "loader resolve" function but got string.`